srv/src/main/java/my/bookshop/handlers/AdminServiceHandler.java

```diff
@@ -56,16 +56,17 @@
 	@Resource(name = AdminService_.CDS_NAME)
 	private DraftService adminService;
 
-	@Autowired
-	private PersistenceService db;
+	private final PersistenceService db;
 
-	@Autowired
-	private Messages messages;
+	private final Messages messages;
 
-	private CqnAnalyzer analyzer;
+	private final CqnAnalyzer analyzer;
+
+	public AdminServiceHandler(DraftService adminService, PersistenceService db, Messages messages, CdsModel model) {
```

Comment:
Is the `@Autowired` annotation on the component constructor optional?

Reply:
exactly :)